src/PJ_rpoly.c

```diff
@@ -34,6 +38,35 @@ static XY s_forward (LP lp, PJ *P) {           /* Spheroidal, forward */
 }
 
 
+static LP s_inverse (XY xy, PJ *P) {           /* Spheroidal, inverse */
+    LP lp = {0.0,0.0};
+    struct pj_opaque *Q = P->opaque;
+    double B, dphi, tp;
+    int i;
+
+    if (fabs(xy.y = P->phi0 + xy.y) <= TOL) {
```

Comment:
> Q->fxa and Q->fxb set in PJ *PROJECTION() and used in the forward calculations. It's confusing and makes it hard to see how the equations translate to code.

I agree, it is very difficult to understand how the code does what the equations says!

> I don't like the do while loop for the Newton-Raphson iteration.

Me too.

> Should I replace the I_ERROR macro with pj_ctx_set_errno?

Don't bother. I think it is inconsistently used in the code and should probably be fixed at some across the whole code base in a dedicated effort. I know @busstoptaktik has been talking about getting rid of the ERROR macros.

> So if I make changes here for code readability, I feel like I should update pj_poly.c as well for consistency. Or do you prefer minimal changes to existing code? Question of philosophy...

That's always a tough one. As long as the changes are backed up by tests I think it adds value to the code. Go ahead :)